doc/barman.md

```diff
@@ -0,0 +1,136 @@
+```
+
+You can also use `additionalMasterReplicationSlots` parameter.
+In this case, when the new master is promoted, the slot will be re-created.
+For example:
+```
+stolonctl --cluster-name=stolon-cluster --store-backend=etcd update --patch '{"additionalMasterReplicationSlots" : ["add_slot"]}'
+```
+
+#### Creating Barman role
+
+```
+bash-4.2$ psql -h 127.0.0.1
+psql (9.6.6)
+postgres=# CREATE ROLE BARMAN LOGIN SUPERUSER REPLICATION PASSWORD 'barman';
+CREATE ROLE
+```
+
+and creating `/var/lib/barman/.pgpass` file on Barman node:
+
+```
+*:*:*:barman:barman
+*:*:*:replica:replica
+```
+
+#### SSH connections
+
+You need to create SSH keys for the Barman and Postgres users. For more details, see [here](http://docs.pgbarman.org/release/2.3/#ssh-connections)
+
+#### Starting replication from Stolon to Barman
+```
+barman receive-wal stolon
+```
+
+#### Check Barman status
+
+```
+$ barman check stolon
+Server stolon:
+        PostgreSQL: OK
+        is_superuser: OK
+        PostgreSQL streaming: OK
+        wal_level: OK
+        replication slot: OK
+        directories: OK
+        retention policy settings: OK
+        backup maximum age: FAILED (interval provided: 3 days, latest backup age: 4 days, 51 minutes, 7 seconds)
+        compression settings: OK
+        failed backups: OK (there are 0 failed backups)
+        minimum redundancy requirements: OK (have 2 backups, expected at least 2)
+        pg_basebackup: OK
+        pg_basebackup compatible: OK
+        pg_basebackup supports tablespaces mapping: OK
+        pg_receivexlog: OK
+        pg_receivexlog compatible: OK
+        receive-wal running: OK
+        archiver errors: OK
+```
+
+### Restoring Stolon cluster from Barman
+
+Barman provides 2 options for restoring PostgreSQL DB:
+
+- Local restoration. At the end of the recovery execution, the selected backup is recovered locally and the destination path contains a data directory ready to be used to start a PostgreSQL instance.
+
+- Remote restoration. Doing this will allow Barman to execute the copy on a remote server using the provided command to connect to the remote host.
+
+You can use both strategies depending on the location of your backup server.
+For remote recovery, you will need to install an additional Barman utility `barman-wal-restore` on all PostgreSQL nodes.
+Here is an example of the remote restoration of Stolon cluster.
+
+```
+stolonctl --cluster-name=stolon-cluster --store-backend=etcd init '{ "initMode": "pitr", "pitrConfig": { "dataRestoreCommand": "ssh barman@<backup-server-host> \"barman recover --remote-ssh-command \\\"ssh postgres@$HOSTNAME\\\" stolon latest /var/lib/pgsql/9.6/postgres\" ", "archiveRecoverySettings": { "restoreCommand": "barman-wal-restore -U barman < backup-server-host > stolon \"%f\" \"%p\"" }}}'
+```
+
+, where
+- < backup-server-host > - Barman address
+- $HOSTNAME - a variable that specifies the node name to be selected as the new primary server.
```

Comment:
where:
`<backup-server-host>`: Barman address
`$HOSTNAME`: a variable that specifies the node name to be selected as the new primary server.